Sources/ArcGISToolkit/Components/Offline/PreplannedListItemView.swift

```diff
@@ -55,6 +75,19 @@ struct PreplannedListItemView: View {
             .font(.body)
     }
     
+    @ViewBuilder private var deleteButton: some View {
+        if model.status.allowsRemoval {
+            Button("Delete") {
+                onDelete()
+                model.removeDownloadedPreplannedMapArea()
+                Task {
+                    await model.load()
```

Comment:
I agree that it is different from the loadable's load method. To me, it is loading a property of the data model, so maybe `model.loadPreplannedMapArea()`? Will leave this open for discussion.

Reply:
> it seems really strange the deleting a local cache would kick off an asynchronous operation that (presumably) goes out over the wire

I kind of agree with your assessment. What I want to achieve with the delete button is this: it deletes the local files for a preplanned map area, then "refresh" the status of the model.

A few factors can affect the new status of the model: network connection is good or not, whether the preplanned map area's packaging status was changed on AGOL. Thus I called the async `load()` method to determine the new status, which I think the is unavoidable.

I thought about calling the `load()` method in the `removeDownloadedPreplannedMapArea()` method but that seems to be overreaching. IMO, removing files from local disk should be a synchronous operation.

Not sure how calling `load()` on-demand would work here - once the delete swipe is done, the list row needs to reflect the latest model status in UI.

Reply:
> But figuring out the current state should not be an async operation.

Because the status of the model is dependent on checking AGOL server to get the states such as `packaging`, `packaged`, `packageFailure`, and `downloading`, it seems it has to be asynchronous? 🤔

Reply:
So the state can't be computed. As for figuring out the state, it all depends on how you model the state. There should be a state for that, perhaps even just `unknown` or more specific `readyForPackaging`, etc. Then moving from one state to the next can't happen without an asynchronous operation (in some cases), but it is the operation itself that should drive the state transition. I hope that is making sense.

Reply:
Thanks for the explanation. Unfortunately, async calls are needed for updating status, as the preplanned are on the server may change to an arbitrary state while the local files remain unchanged. Thus, it would be safer to fetch the latest state when the local files are removed.

> but it is the operation itself that should drive the state transition

This totally makes sense! Will move the state transition within the operation.

Reply:
Thanks. I still don't understand why deleting a cache needs to kick off a network request and/or why the state can't just go to `packaged`, but clearly you understand this more than I do, so I'll take your word for it that there isn't another way.